Automattic/jetpack
# Tools: Cleanup and update release scripts

#### Changes proposed in this Pull Request:

* Remove projects/plugins/jetpack/tools/prepare-build-branch.sh

  Seems to have existed for action automattic/action-jetpack-build-to-branch, which we don't use anymore.

* Rm tools/deploy-to-master-stable-branch.sh
    
  We used to maintain a "master-stable" branch, which was a lightly curated "alpha" for people to test with. But we haven't pushed to it since 2017, so let's get rid of it.

* Update branches in GH workflows
    
  Remove master-stable, rename release branch pattern for new style.

* Rm tools/git-hooks/pre-commit
    
  Empty since 2016, and apparently unused in favor of tools/pre-commit-hook.js.

* Move actual git hooks into tools/git-hooks
    
  Seems like sensible organization.

* Rm tools/trigger-travis-build.sh
    
  We don't use Travis anymore.

* Replace get-version.sh and version-update.sh
    
  version-update.sh was very Jetpack-specific, and tried to do a bit too much (prompting, committing to git) for my tastes.
    
  get-version.sh was already reasonably generic, but could benefit from sharing more logic than the normalize function with version-update.sh.
    
  So I wrote a new plugin-version.sh, partially based on version-update.sh, to handle the operations of "get plugin version" and "update plugin version" generically for any monorepo plugin.

* Rewrite version-packages.sh
    
  I started out by making it more generic, and wound up rewriting it almost entirely to be able to pass arbitrary args on to Composer, to read the monorepo package names more directly, and to update all the packages at once instead of one at a time (which can cut down on the need to do --no-update or the like).

* Rewrite build-release-branch.sh as create-release-branch.sh
    
  The build is supposed to be done by GitHub Actions now, so a lot of the former logic here can go away. All we need to do is create and push the branch with the proper naming.
    
  At the same time, new logic was needed to build generically. And I included a bit more in the way of safety checks while I was at it.

* Rewrite deploy-to-svn.sh
    
  To pull from the mirror repo, and to get the WordPress.org plugin name from composer like everything else.

#### Jetpack product discussion


1199714989172614-as-1199714989172643

#### Does this pull request change what data or activity we track or use?


No.

#### Testing instructions:






* Try out all the scripts, see if they do the right thing.
* Check to make sure I didn't leave out any steps we need in create-release-branch.sh.

#### Proposed changelog entry for your changes:


* None needed.
